skimage/measure/_regionprops.py

```diff
@@ -835,6 +846,114 @@ def regionprops(label_image, intensity_image=None, cache=True):
+        are considered as the object.
+    neighbourhood : 4 or 8 for 2D images, 6 or 26 for 3D images, optional
+        Neighborhood connectivity for object determination. If object is 
+        4-connected, then background is 8-connected, and conversely.
+        8-connectivity is used by default for 2D images.
+        26-connectivity is used by default for 3D images.
+
+    Returns
+    -------
+    euler_number : int
+        Euler characteristic
+
+    References
+    ----------
+    .. [1] S. Rivollier. Analyse d’image geometrique et morphometrique par 
+           diagrammes de forme et voisinages adaptatifs generaux. PhD thesis,
+           2010. Ecole Nationale Superieure des Mines de Saint-Etienne.
+           https://tel.archives-ouvertes.fr/tel-00560838
+    .. [2] Ohser J., Nagel W., Schladitz K. (2002) The Euler Number of 
+           Discretized Sets - On the Choice of Adjacency in Homogeneous 
+           Lattices. In: Mecke K., Stoyan D. (eds) Morphology of Condensed 
+           Matter. Lecture Notes in Physics, vol 600. Springer, Berlin, 
+           Heidelberg.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> SAMPLE = np.zeros((100,100,100));
+    >>> SAMPLE[40:60, 40:60, 40:60]=1
+    >>> euler_number(SAMPLE) # doctest: +ELLIPSIS
+    1...
+    >>> SAMPLE[45:55,45:55,45:55] = 0;
+    >>> euler_number(SAMPLE) # doctest: +ELLIPSIS
+    2...
+    >>> SAMPLE = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+    ...                    [1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0],
+    ...                    [0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1],
+    ...                    [0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1]])
+    >>> euler_number(SAMPLE)  # doctest: +ELLIPSIS
+    0...
+    >>> euler_number(SAMPLE, neighbourhood=4)  # doctest: +ELLIPSIS
+    2...
+    """
+
+    # as image can be a label image, transform it to binary
+    image = (image > 0).astype(np.int)
+    image = pad(image, ((1, 1),), mode='constant')
+
+    if np.ndim(image) == 2:
+
+        F = np.array([[0, 0, 0], [0, 1, 4], [0, 2, 8]])
+        if neighbourhood == 4:
+            coefs = [0,  1,  0,  0,  0,  0,  0, -
+                     1,  0,  1,  0,  0,  0,  0,  0,  0]
+        else:
+            coefs = [0,  0,  0,  0,  0,  0, -1,
+                     0,  1,  0,  0,  0,  0,  0, -1,  0]
+        bins = 16
+    else:  # 3D images
+        F = np.array([[[0, 0, 0], [0, 0, 0], [0, 0, 0]],
```

Comment:
is it possible to find a more self-explaining name than `F`?

Reply:
I changed it to config, as this describes a configuration neighbourhood.